tests/unit/dogstatsd/test_container.py

```diff
@@ -125,12 +127,57 @@ def get_mock_open(read_data=None):
         ),
     ),
 )
-def test_container_id(file_contents, expected_container_id):
+def test_container_id_from_cgroup(file_contents, expected_container_id):
     with get_mock_open(read_data=file_contents) as mock_open:
         if file_contents is None:
             mock_open.side_effect = IOError
 
-        reader = ContainerID()
+        with mock.patch("os.stat", mock.MagicMock(return_value=mock.Mock(st_ino=0xEFFFFFFB))):
+            reader = Cgroup()
         assert expected_container_id == reader.container_id
 
         mock_open.assert_called_once_with("/proc/self/cgroup", mode="r")
+
+
+def test_container_id_inode():
+    """Test that the inode is returned when the container ID cannot be found."""
+    with mock.patch("datadog.dogstatsd.container.open", mock.mock_open(read_data="0::/")) as mock_open:
+        with mock.patch("os.stat", mock.MagicMock(return_value=mock.Mock(st_ino=1234))):
+            reader = Cgroup()
+        assert reader.container_id == "in-1234"
+        mock_open.assert_called_once_with("/proc/self/cgroup", mode="r")
+
+    cgroupv1_priority = """
+12:cpu,cpuacct:/
+11:hugetlb:/
+10:devices:/
+9:rdma:/
+8:net_cls,net_prio:/
+7:memory:/
+6:cpuset:/
+5:pids:/
+4:freezer:/
+3:perf_event:/
+2:blkio:/
+1:name=systemd:/
+0::/
+"""
+
+    def inode_stat_mock(path):
+        global MEMORY_PRIORITY_VALID
+        if not MEMORY_PRIORITY_VALID:
+            # Priority is given to the cgroupv1 controller.
+            assert path == "/sys/fs/cgroup/memory/"
```

Comment:
I may be misreading things again, but wouldn't this always fail? The first stat is on `CGROUP_NS_PATH`.

Reply:
Yes I think it's because `pytest` actually just want an `assert` to be validated at some point in a run. I found another way to do this:
```
tests/unit/dogstatsd/test_container.py::test_container_id_from_cgroup[\n11:hugetlb:/ecs/55091c13-b8cf-4801-b527-f4601742204d/432624d2150diff --git a/tests/unit/dogstatsd/test_container.py b/tests/unit/dogstatsd/test_container.py
index cf259ae..f8da40e 100644
--- a/tests/unit/dogstatsd/test_container.py
+++ b/tests/unit/dogstatsd/test_container.py
@@ -12,7 +12,7 @@ import pytest

 from datadog.dogstatsd.container import Cgroup

-MEMORY_PRIORITY_VALID = False
+MEMORY_PRIORITY_VALID = []


 def get_mock_open(read_data=None):
@@ -165,10 +165,7 @@ def test_container_id_inode():

     def inode_stat_mock(path):
         global MEMORY_PRIORITY_VALID
-        if not MEMORY_PRIORITY_VALID:
-            # Priority is given to the cgroupv1 controller.
-            assert path == "/sys/fs/cgroup/memory/"
-            MEMORY_PRIORITY_VALID = True
+        MEMORY_PRIORITY_VALID.append(path)

         # The cgroupv1 controller is mounted on inode 0. This will cause a fallback to the cgroupv2 controller.
         if path == "/sys/fs/cgroup/memory/":
@@ -180,4 +177,9 @@ def test_container_id_inode():
         with mock.patch("os.stat", mock.MagicMock(side_effect=inode_stat_mock)):
             reader = Cgroup()
         assert reader.container_id == "in-1234"
+        global MEMORY_PRIORITY_VALID
+        assert MEMORY_PRIORITY_VALID[-2:] == [
+            "/sys/fs/cgroup/memory/",
+            "/sys/fs/cgroup/"
+        ]
         mock_open.assert_called_once_with("/proc/self/cgroup", mode="r")
```

Note that `MEMORY_PRIORITY_VALID` will contain at first two `CGROUP_NS_PATH`. One for the first `os.stat()`, the other one for the `os.path.exist()` since it's actually also using an `os.stat()`.

Good catch 👍